### PR TITLE
kraken: make data copyable and add method on data_manager for handling disruptions

### PR DESCRIPTION
**Pull request by kinnou02**
We serialize in a string then we deserialize it, it's ugly, but it
will do the job for now. The ClonningArchive will save us!!!

### CODE REVIEW

File: source/type/data.cpp [hunk 1]
```diff
@@ -105,11 +105,23 @@ void Data::load_lz4(const std::string & filename) {
...
 
+void Data::save(std::stringstream& ss) {
+    boost::archive::binary_oarchive oa(ss);
+    oa << *this;
+}
+
+void Data::load(std::stringstream& ss) {
+    boost::archive::binary_iarchive ia(ss);
+    this->build_raptor();
+    ia >> *this;
```
Comment (TeXitoi) [inline source/type/data.cpp]: build_raptor is at the wrong place

File: source/type/data.cpp [hunk 2]
```diff
@@ -105,11 +105,23 @@ void Data::load_lz4(const std::string & filename) {
...
+}
+
```
Comment (TeXitoi) [inline source/type/data.cpp]: I think it can be better to have load and save on a std::[io]stream, and sharing its usage with the serialization/deserialization on lz4

File: source/kraken/data_manager.h [hunk 3]
```diff
@@ -56,12 +64,20 @@ class DataManager{
...
+        {
+            std::stringstream ss;
+            current_data->save(ss);
+            auto data = std::make_shared<Data>();
+            data->load(ss);
+            //call the required method here
+            std::swap(current_data, data);
+        }
+        release_memory();
+    }
```
Comment (TeXitoi) [inline source/kraken/data_manager.h:78]: I prefere that, but that's a question of taste:

``` c++
template<typename Disruption>
void apply_disruptions(const Disruption& /*disruptions*/){
    std::stringstream ss;
    current_data->save(ss);
    auto data = std::make_shared<Data>();
    data->load(ss);
    //call the required method here
    current_data = std::move(data);
    release_memory();
}
```

File: source/kraken/data_manager.h [hunk 4]
```diff
@@ -56,12 +64,20 @@ class DataManager{
...
+    void apply_disruptions(const Disruption& /*disruptions*/){
+        {
+            std::stringstream ss;
+            current_data->save(ss);
+            auto data = std::make_shared<Data>();
+            data->load(ss);
+            //call the required method here
+            std::swap(current_data, data);
+        }
+        release_memory();
```
Comment (TeXitoi) [inline source/kraken/data_manager.h:77]: Note that, if another thread is still using the old data, the memory will not be freed.

**Comment by kinnou02** [comment]
jenkins, test this please

File: source/type/data.h [hunk 5]
```diff
@@ -120,28 +120,34 @@ class Data : boost::noncopyable{
 
     friend class boost::serialization::access;
 
-    bool last_load = true;
+    std::atomic<bool> last_load;
```
Comment (TeXitoi) [inline source/type/data.h]: this modification seems irrelevant.

**Comment by TeXitoi** [comment]
OK for me after resolving merge conflicts.

**Comment by kinnou02** [comment]
jenkins, test this please
